scripts/deploy.py

```diff
@@ -0,0 +1,497 @@
+    UpdateChangeLog(p)
+    p.AddStep(WriteAndCommitVersion('dev-version', add_tag=False))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout release"))
+    p.AddStep(RunCmdStep('git merge release', doc="Merge release into master"))
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(WriteAndCommitVersion('new-version', add_tag=True))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.AddStep(
+        RunCmdStep(
+            'git push --all origin', doc="Push release changes to github"))
+    p.AddStep(RunCmdStep('git push --tags origin', doc="Push tags to github"))
+    p.Run('new-release')
+
+
+@cli.command()
+@click.pass_context
+def cherrypick(ctx):
+    p = ctx.obj['pipeline']
+
+    CheckOwnIntegrityAndFetch(p)
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(SelectCommitToCherrypick(jump_on_end=3))
+    p.AddStep(CherryPick)
+    p.AddStep(RunJump(-2))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.Run('cherrypick')
+
+
+@cli.command()
+@click.pass_context
+def new_patch_release(ctx):
+    p = ctx.obj['pipeline']
+    CheckOwnIntegrityAndFetch(p)
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(ReadCurrentVerionFromFile)
+    p.AddStep(GetNextPatchVersion)
+    UpdateChangeLog(p)
+    p.AddStep(WriteAndCommitVersion('new-version', add_tag=True))
+    p.AddStep(
+        RunCmdStep(
+            'git push --all origin', doc="Push release changes to github"))
+    p.AddStep(RunCmdStep('git push --tags origin', doc="Push tags to github"))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.Run('new-patch')
+
+
+def CheckOwnIntegrityAndFetch(p):
+    p.AddStep(RunStoreFileSha(OWN_FILENAME))
+    p.AddStep(
+        RunCmdStep(
+            'git diff-index --quiet HEAD --',
+            doc="Check that git doesn't have uncommited changes"))
+    p.AddStep(RunCmdStep('git checkout release', doc="Checkout release"))
+    p.AddStep(RunCmdStep('git pull', doc="Pull release from upstream"))
+    p.AddStep(RunCmdStep('git checkout master', doc="Checkout master"))
+    p.AddStep(RunCmdStep('git pull', doc="Pull master from upstream"))
+    p.AddStep(CompareFileSha(OWN_FILENAME, require_equal=True))
+
+
+def UpdateChangeLog(p):
+    p.AddStep(SaveGitLog)
+    p.AddStep(PrependGitLogToChangeLog)
+    p.AddStep(RunStoreFileSha(CHANGELOG_PATH))
+    p.AddStep(
+        RunCmdStep(
+            '%s %s' % (os.getenv('EDITOR', 'vim'), CHANGELOG_PATH),
+            doc="Edit changelog"))
+    p.AddStep(CompareFileSha(CHANGELOG_PATH, require_equal=False))
+    p.AddStep(RunCmdStep('git add %s' % CHANGELOG_PATH))
+
+
+############################################################################
+# Pipeline machinery
+############################################################################
+
+
+def RunJump(where):
```

Comment:
See below also, but "steps *forward*" indicates this should be named `offset` or similar, not `where`, which implies an absolute target, not a relative one

Reply:
Done.